pkg/serverless/trace/propagation/carriers.go

```diff
@@ -164,10 +165,22 @@ func snsSqsMessageCarrier(event events.SQSMessage) (tracer.TextMapReader, error)
 // snsEntityCarrier returns the tracer.TextMapReader used to extract trace
 // context from the attributes of an events.SNSEntity type.
 func snsEntityCarrier(event events.SNSEntity) (tracer.TextMapReader, error) {
-	msgAttrs, ok := event.MessageAttributes[datadogSQSHeader]
-	if !ok {
-		return nil, errorNoDDContextFound
+	// Check if this is a regular SNS message with Datadog trace information
+	if msgAttrs, ok := event.MessageAttributes[datadogTraceHeader]; ok {
+		return handleRegularSNSMessage(msgAttrs)
+	}
+
+	// If not, check if this is an EventBridge event sent through SNS
+	var eventBridgeEvent map[string]interface{}
+	if err := json.Unmarshal([]byte(event.Message), &eventBridgeEvent); err == nil {
```

Comment:
Why not unmarshall into an `events.EventBridgeEvent` then use the `eventBridgeCarrier` method below so you do not have to reinvent the wheel?

Reply:
Yes you're exactly right. I was totally overthinking this. 

After making these changes, the function is practically identical to my existing `eventBridgeCarrier()`. So we can just get rid of `handleEventBridgeThroughSNS()` entirely and replace it with `eventBridgeCarrier()`.